backend/ttrs/views.py

```diff
@@ -53,11 +56,10 @@ class StudentCreate(generics.CreateAPIView):
 class StudentDetail(generics.RetrieveUpdateDestroyAPIView):
     queryset = Student.objects.all()
     serializer_class = StudentSerializer
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsAuthenticated, IsStudent)
```

Comment:
Indeed, I concerned that problem but I concentrated to implementation of TimeTableAPIViews.
So, I agree with your opinion, I'll change the permissions.
And, the `StudentList` have permission of `IsAuthenticated' only. I think the permission `IsAdminUser` should be added to there.

Reply:
The `StudentDetail` APIView cannot be accessed by other student because the view overrode `get_object` method and this method will return the information of requester. So, `IsStudent` permission is enough.